core/conversion/evaluators/prim.cpp

```diff
@@ -101,6 +102,28 @@ auto prim_registrations =
                         return c10::optional<torch::jit::IValue>(std::move(torch::jit::IValue(list)));
                       }
                     }})
+        .evaluator({c10::Symbol::fromQualString("prim::dtype"),
+                    [](const torch::jit::Node* n, kwargs& args) -> c10::optional<torch::jit::IValue> {
+                      auto input = args.at(n->input(0));
+                      if (input.isITensor()) {
+                        auto trt_dtype = input.ITensor()->getType();
+                        return static_cast<int>(util::TRTDataTypeToScalarType(trt_dtype));
+                      } else if (input.isIValue()) {
+                        if (input.IValue()->isTensor()) {
+                          auto pyt_input = input.IValue()->toTensor();
+                          return static_cast<int>(c10::typeMetaToScalarType(pyt_input.dtype()));
```

Comment:
Any reason to not use pyt_input.scalar_type() directly? would they do different things sometimes?

Reply:
done